privacyidea/lib/tokens/pushtoken.py

```diff
@@ -860,8 +860,10 @@ def create_challenge(self, transactionid=None, options=None):
                 db_challenge.save()
                 self.challenge_janitor()
 
-            # If sending the Push message failed, we still raise an error.
+            # If sending the Push message failed, we still raise an error and a warning.
             if not res:
+                log.warning("Failed to submit message to firebase service for token {0!s}."
+                            .format(self.token.serial))
```

Comment:
I am sorry for my late comment.
I was not quite sure about this.

But in theory a serial number could contain a special character or a German umlaut. If we print it in the log message this way, we will raise an exception.
So we should do

    log.warnung(u"Failed to .....".format())

To allow logging unicode.

Thanks a lot

Reply:
This should be fixed now.